tests/scripts/all.sh

```diff
@@ -945,6 +945,39 @@ helper_libtestdriver1_make_main() {
     make CC=$ASAN_CC CFLAGS="$ASAN_CFLAGS -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS" "$@"
 }
 
+# $1: target which can be "client" or "server"
+helper_crypto_client_build() {
+    TARGET=$1
+    TARGET_LIB=libpsa$TARGET
+
+    cp $CONFIG_H $CONFIG_H.bak
```

Comment:
It this not done by the `pre_back_up`  ref [1](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L602) and [2](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L6284)? 

What happens if make command fails, or if the config file is the [driver config](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L140)?

Reply:
> It this not done by the pre_back_up ref [1](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L602) and [2](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L6284)?

Yes, but `pre_back_up` dumps files which are restore at the end of `run_component`. In our case, however, we are modifying config files twice in the same component. So unless I missed something here, we need this extra logic. Wdyt?

> What happens if make command fails

Right, we might end up with a modified config file which can erroneously be picked up as the valid one on the next run (unless the user realizes that and clears the status with `git restore ...`). I will fix this!

> if the config file is the [driver config](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh#L140)

Mmmm right, I didn't thought about this but I need to make a copy also of this file! Thanks for the notification